modules/transactionpool/persist.go

```diff
@@ -95,9 +115,22 @@ func (tp *TransactionPool) initPersist() error {
 		if resetErr != nil {
 			return resetErr
 		}
-		return tp.consensusSet.ConsensusSetSubscribe(tp, modules.ConsensusChangeBeginning)
+		freshScanErr := tp.consensusSet.ConsensusSetSubscribe(tp, modules.ConsensusChangeBeginning)
+		if freshScanErr != nil {
+			return freshScanErr
+		}
+		tp.tg.OnStop(func() {
+			tp.consensusSet.Unsubscribe(tp)
+		})
+		return nil
```

Comment:
typically I write this like:
```go
err = tp.consensusSet.ConsensusSetSubscribe(tp, modules.ConsensusChangeBeginning)
```
so that control flow can then proceed to the standard error handling. Avoids code duplication, although I can see it being a little surprising.